### ISSUE TITLE
Parser::default breaks --help for subcommands

### ISSUE THREAD

**Issue by 0x5a4** [pacak/bpaf#21]
I found myself wanting to implement an App that has sub-commands but also does 
something when run completely without options. This is possible using `Parser::default`, but
using it breaks `--help` for subcommands. Instead of printing the help for the corresponding subcommand, the Help-page for the complete App is printed no matter what...

**Comment by pacak** [comment]
Do you have any code to look at?

**Comment by pacak** [comment]
Okay, managed to reproduce something... Let me see what's going on.

FWIW intended way to fallback to a default subcommand is doing something like this:

```rust
    let opt = Info::default()
        .descr("The stupid content tracker")
        .for_parser(
            fetch_cmd
                .or_else(add_cmd)
                .or_else(Parser::pure(Opt::Otherwise)),
        )
        .run();
```

For the full code you can take a look at `git.rs`, I added `Opt::Otherwise` enum variant and changed ^ code.

**Comment by pacak** [comment]
https://github.com/pacak/bpaf/commit/9f4e10d58de9ac57365777d1791e6297c268988b

Released 0.4.9, please reopen if problem persists.
